consul/prepared_query_endpoint.go

```diff
@@ -562,6 +572,18 @@ func tagFilter(tags []string, nodes structs.CheckServiceNodes) structs.CheckServ
 	return nodes[:n]
 }
 
+// nodeMetaFilter returns a list of the nodes who satisfy the given metadata filters. Nodes
+// must have ALL the given tags.
+func nodeMetaFilter(filters map[string]string, nodes structs.CheckServiceNodes) structs.CheckServiceNodes {
```

Comment:
Since this can only reduce the original slice I wonder if it's worth doing an in-place algorithm like https://github.com/hashicorp/consul/blob/master/consul/acl.go#L359-L373. Would be curious what benchmarks show for this.